lib/dl_query_processing/dl_query_processing/multi_query/splitters/query_fork.py

```diff
@@ -426,11 +455,21 @@ def get_split_masks(
             for extract, alias in sorted(extract_to_alias_map.items(), key=lambda t: t[0].complexity, reverse=True):
                 joining_node = self._normalize_joining_node(joining_node=joining_node, extract=extract, alias=alias)
 
+            add_filters = []
+
             # Collect indices of filters that should be applied to the sub-query
             filter_indices: set[int] = set()
             for filter_idx, filter_formula in enumerate(query.filters):
                 if filter_formula.original_field_id in qfork_info.bfb_field_ids:
-                    # Filter field is in BFB, so exclude it.
+                    # Filter field is in BFB, so exclude it unless it is mutated by one of the BFB mutations
+                    if any(
+                        contains_node(filter_formula.formula_obj, mutation.original)
```

Comment:
imho, a better way to check if mutation has occured is https://github.com/datalens-tech/datalens-backend/blob/aa8ebfd5b857d011ce2e2909d6a69a105e821441/lib/dl_query_processing/dl_query_processing/compilation/query_mutator.py#L258

this way we don't need `contains_node` at all